linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/tests/pam_tally2_absent_account_config.fail.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+# platform = multi_platform_sle,Ubuntu 20.04
+
+cp -f common-account_pam_tally2_absent /etc/pam.d/common-account
```

Comment:
I would suggest to prepare the `/etc/pam.d/common-account` and `/etc/pam.d/common-auth` programmatically in the test scenario scripts themselves instead of keeping example files in the project and copying them. It would be more scalable and would reduce the number of files in the project. Could you consider this option, please? 

Also valid for the `accounts_passwords_pam_tally2_unlock_time` rule.

Reply:
:bow: Done in e3f3597

Reply:
I meant to remove the following files:
- common-account_pam_tally2_absent
- common-account_pam_tally2_configured
- common-auth_pam_tally2_deny_missing
- common-auth_pam_tally2_even_deny_root_configured
- common-auth_pam_tally2_even_deny_root_missing

These 5 files have between 3 and 5 uncommented lines. I suggested to create the content of `/etc/pam.d/common-account` and `/etc/pam.d/common-auth` directly in the test scripts instead of overwriting them by copying these 5 "templated" files.

Your approach is not wrong but I believe these files are unnecessary. Also there are many commented lines there which are not relevant for testing purposes. If you remove them and generate the content of the PAM files directly in the test scenario scripts (by using echo "line..." >> pam_file, for example), the readability should be improved, making it easier to understand the test scenarios once only a single file would give all the context. It would also be more flexible to maintain the test scenarios and easier to include more test scenarios whenever necessary.

Reply:
Change for all new tests the approach to use dynamically generated file contents instead of file copies.